src/main/java/hwr/oop/riddler/model/component/CellGroup.java

```diff
@@ -1,4 +1,17 @@
 package hwr.oop.riddler.model.component;
 
-public interface CellGroup {
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public abstract class CellGroup {
+    protected Set<Cell> cells = new HashSet<>();
```

Comment:
Das Problem hierbei ist vielleicht, dass dadurch das wir es hier als Set speichern, die Reihenfolge verloren geht. Also in einer Row hat jede Cell ja trotzdem noch eine Position.

Vielleicht cells garnicht in CellGroup speichern, dafür aber eine abstrakte Methode hinzufügen, die die Implementationen der Klasse dazu zwingt ein Set mit cells zurückzugeben? So kann man die Cells noch geordnet in der Implemantation der Cellgroup speicher, jedoch wird garantiert, dass jede Implemntation die Cells zurückgeben kann.

Reply:
Weiß aber garnicht, wie man das am besten machen würde. Vielleicht ist die Reihenfolge in den Rows/Columns letzendnlich auch egal, bin mir gerade wirklich nicht sicher. + Das kann man ja alles auch in der Zukunft nochmal umbauen.

Reply:
Ja, ich bin mir da auch nicht so sicher. Wir können es ja später umbauen je nach dem was wir so brauchen. Ich lass es erstmal so.